smart_open/tests/test_s3.py

```diff
@@ -672,6 +672,14 @@ def test_writebuffer(self):
 
             assert actual == contents
 
+    def test_str(self):
+        """Check the stringifying works!"""
+        fout = smart_open.s3.open(BUCKET_NAME, 'key', 'wb', multipart_upload=False)
+        fout.write(b"text")
+        fout.flush()
+        fout.close()
+        assert str(fout) != ''
```

Comment:
Why are we checking against the empty string here? Why not just check against what the expected value should be?

Reply:
Also, is writing, flushing, closing explicitly necessary? If not, then it shouldn't be part of the test, and a context manager will be enough.